src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +140,33 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
```

Comment:
Why not make it a case in `WriteObjectValue`?

Reply:
Object is Json Object, but here the BinaryData can ben raw string or raw binary

Reply:
I think there are two reasons that we don't want to make it a case in `WriteObjectValue`:
1. In mgmt plane, we explicitly know that the input parameter is a BinaryData, so we don't need the type check in `WriteObjectValue`
2. `WriteObjectValue` already has a case about `BinaryData`, which writes the value as Base64 encoded json string, I don't think we want to break current algorithm.